src/gax-internal/tests/http_retry_loop.rs

```diff
@@ -103,16 +117,53 @@ mod tests {
             .await;
         let response = response?.into_body();
         assert_eq!(response, json!({"status": "done"}));
+
+        let spans = TestLayer::capture(&guard);
+        if enable_tracing {
+            assert_eq!(spans.len(), 3, "Expected 3 spans");
+            // Span 1 (attempt 0)
+            assert_eq!(spans[0].name, "http_request");
+            assert!(
+                !spans[0]
+                    .attributes
+                    .contains_key(semconv::HTTP_REQUEST_RESEND_COUNT)
+            );
+            // Span 2 (attempt 1)
+            assert_eq!(spans[1].name, "http_request");
+            assert_eq!(
+                spans[1].attributes.get(semconv::HTTP_REQUEST_RESEND_COUNT),
+                Some(&"1".to_string())
+            );
+            // Span 3 (attempt 2)
+            assert_eq!(spans[2].name, "http_request");
+            assert_eq!(
+                spans[2].attributes.get(semconv::HTTP_REQUEST_RESEND_COUNT),
+                Some(&"2".to_string())
+            );
+        } else {
+            assert_eq!(spans.len(), 0, "Expected 0 spans when tracing is disabled");
+        }
         Ok(())
     }
 
+    #[test_case(false; "tracing off")]
+    #[test_case(true; "tracing on")]
     #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-    async fn retry_loop_too_many_transients() -> Result<()> {
+    async fn retry_loop_too_many_transients(enable_tracing: bool) -> Result<()> {
+        if enable_tracing && !cfg!(feature = "_unstable-o12y") {
```

Comment:
tests with a lot of logic like this are better off split into two tests. Feel free to refactor the common initialization to a helper function in the `mod tests { ... }` module.